plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -948,6 +937,8 @@ private void finishOptimize(ConnectorSession session, IcebergTableExecuteHandle
         }
         RewriteFiles rewriteFiles = transaction.newRewrite();
         rewriteFiles.rewriteFiles(scannedFiles, newFiles);
+        optimizeHandle.getSnapshotId().map(icebergTable::snapshot)
+                .ifPresent(snapshot -> rewriteFiles.validateFromSnapshot(snapshot.snapshotId()));
```

Comment:
what's the case where `optimizeHandle.getSnapshotId()` is empty?

is it only for an initially empty table that didn't have any snapshot yet?

Reply:
I couldn't find such case. Even empty table has snapshot id as far as I confirmed.

Reply:
if so, let's make `optimizeHandle.getSnapshotId().` not an  Optional

Reply:
indeed, it would be hard to analyze a table that doesn't exist yet :)